tests/unit/dispensary/test.updater.js

```diff
@@ -68,135 +68,75 @@ describe(__filename, () => {
-    });
-
-    const fakeConsole = { error: () => {}, log: () => {} };
-    const fakeFS = {
-      writeFile: (filename, contents, options, callback) => {
-        callback(new Error('Fail!'));
-      },
-    };
-
-    sinon.stub(dispensary, '_getCachedHashes').callsFake(() => {
-      return [];
-    });
-
-    return dispensary
-      .updateCommand([], fakeConsole, fakeFS)
-      .then(unexpectedSuccess)
-      .catch((err) => {
-        // eslint-disable-next-line jest/no-conditional-expect
-        expect(err.message).toEqual('UpdateError: Error: Fail!');
-        // eslint-disable-next-line jest/no-conditional-expect
-        expect(err).toBeInstanceOf(Error);
-      });
-  });
-
   it('should return an array of hashes', () => {
-    const dispensary = new Updater({
-      libraries: TEST_LIBRARIES_JSON_PATH,
-    });
+    const updater = new Updater({ libraryFile: TEST_LIBRARIES_JSON_PATH });
     const fakeConsole = { error: () => {}, log: () => {} };
 
-    sinon.stub(dispensary, '_getCachedHashes').callsFake(() => {
+    sinon.stub(updater, '_getCachedHashes').callsFake(() => {
       return [];
     });
 
-    return dispensary.run(fakeConsole).then((hashes) => {
-      expect(hashes.length).toBe(20);
+    return updater.run(fakeConsole).then((hashes) => {
+      // 14 = 2 * 4 versions for backbone + 1 * 6 versions for
+      // backbone.localStorage (because there is no minified filename for it)
+      expect(hashes.length).toBe(14);
       expect(hashes).toBeInstanceOf(Array);
     });
   });
 
   it('should set files', () => {
-    const dispensary = new Updater({
-      libraries: TEST_LIBRARIES_JSON_PATH,
-    });
+    const updater = new Updater({ libraryFile: TEST_LIBRARIES_JSON_PATH });
 
-    return dispensary
+    return updater
       .getLibraries()
       .then((libraries) => {
-        return dispensary.getFiles(libraries);
+        return updater.getFiles(libraries);
       })
       .then((libraries) => {
         expect(libraries[0].name).toEqual('backbone');
-        expect(libraries[0].files.length).toBe(24);
+        expect(libraries[0].files.length).toBe(8);
+        // No `filenameMinified` for this lib.
+        expect(libraries[1].name).toEqual('backbone.localStorage');
+        expect(libraries[1].files.length).toBe(6);
       });
   });
 
   it('should set hashes', () => {
-    const dispensary = new Updater({
-      libraries: TEST_LIBRARIES_JSON_PATH,
-    });
+    const updater = new Updater({ libraryFile: TEST_LIBRARIES_JSON_PATH });
 
-    return dispensary
+    return updater
       .getLibraries()
       .then((libraries) => {
-        return dispensary.getFiles(libraries);
+        return updater.getFiles(libraries);
       })
       .then((libraries) => {
-        return dispensary.getHashes(libraries);
+        return updater.getHashes(libraries);
       })
       .then((libraries) => {
-        expect(libraries[0].files.length).toBe(32);
+        expect(libraries[0].files.length).toBe(8);
```

Comment:
Why did this change from 32 to 8?

Reply:
I think this was a bug. Reading the code, I don't understand how we could have 32 files given that the fixtures file contains 4 versions for `backbone` with two possible files (minified and "normal"). I think, previously, test cases were leaking because skipping some changed these numbers as well.. That's related to the `_files` hack I believe.